internal/trigger/http/rest.go

```diff
@@ -116,7 +117,15 @@ func (trigger *Trigger) requestHandler(writer http.ResponseWriter, r *http.Reque
 
 	if messageError != nil {
 		// Process logs the error, so no need to log it here.
-		writer.WriteHeader(http.StatusInternalServerError)
+		if edgexErrors.Kind(messageError) == edgexErrors.KindContractInvalid {
+			if trigger.serviceBinding.RawInput() {
+				writer.WriteHeader(http.StatusUnprocessableEntity)
```

Comment:
Since we received `KindContractInvalid` we can assume just `StatusBadRequest` and not need .RawInput() , thus avoiding then API and doc updates.

Reply:
I sort of agree.  The bug reported did specifically mention status code 422 so wanted to keep that support.  Going to take a quick look at internalizing the check while I have a minute.